de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ValidatorTest.java

```diff
@@ -0,0 +1,291 @@
+		
+		// Check that there is an error icon in the document table
+		SwtBotSection documentSection = getSWTBotSection(Document.class);
+		SWTBotTable documentTable = documentSection.getSWTBotTable();
+		SWTBotTableItem documentNameItem = documentTable.getTableItem(0).click(0);
+		
+		Image actualIcon = getImageForTableItem(documentNameItem);
+		Image expectedIcon = new MarkerImageProvider(null).getProblemImageForSeverity(IMarker.SEVERITY_ERROR);
+		assertEquals(expectedIcon, actualIcon);
+		
+		// Unexpand the document section
+		documentSection.click();
+		
+		// Check that clicking the header link collapses all sections which have no error (i.e. all except the document section)
+		// and expands those with an error (i.e. the document section)
+		final String EXPEDTED_ERROR = DvlmNamingConventionValidator.WARNING_PREFIX 
+				+ ConfigurationTree.class.getSimpleName() + "."
+				+ ElementConfiguration.class.getSimpleName() + ".."
+				+ DvlmNamingConventionValidator.WARNING_DOTS_SUFFIX;
+		SwtBotHyperlink swtBotHyperlink = getSWTBotHyperlink(EXPEDTED_ERROR);
+		swtBotHyperlink.click();
+		
+		List<SwtBotSection> expandedSwtBotSections = getExpandedSections();
+		assertEquals(1, expandedSwtBotSections.size());
+		assertSameWidget(documentSection.widget, expandedSwtBotSections.get(0).widget);
+		
+		// Check that the header label has the right tooltip text
+		String tooltip = swtBotHyperlink.getToolTipText();
+		assertEquals(EXPEDTED_ERROR, tooltip);
+	}
+	
+	/**
+	 * Gets the currently expanded sections
+	 * @return a list containing only expanded sections
+	 */
+	private List<SwtBotSection> getExpandedSections() {
+		List<? extends Section> sections = bot.getFinder().findControls(widgetOfType(Section.class));
+		List<SwtBotSection> expandedSwtBotSections = sections.stream()
+				.map(SwtBotSection::new)
+				.filter(SwtBotSection::isExpanded)
+				.collect(Collectors.toList());
+		return expandedSwtBotSections;
+	}
+	
+	/**
+	 * Gets the warnings entry from the problem view
+	 * @param countExpectedWarnings the expected number of warnings
+	 * @return the warning entry
+	 */
+	private SWTBotTreeItem getWarnings(int countExpectedWarnings) {
+		String plural = (countExpectedWarnings > 1) ? "s" : "";
+		return problemView.bot().tree().getTreeItem("Warnings (" + countExpectedWarnings + " item" + plural + ")");
+	}
+	
+	/**
+	 * Gets the error entry from the problem view
+	 * @param countExpectedErrors the expected number of errors
+	 * @return the warning entry
+	 */
+	private SWTBotTreeItem getErrors(int countExpectedErrors) {
+		String plural = (countExpectedErrors > 1) ? "s" : "";
+		return problemView.bot().tree().getTreeItem("Errors (" + countExpectedErrors + " item" + plural + ")");
+	}
+	
+	/**
+	 * Gets the image of a table item
+	 * @param swtBotTableItem the swtbot table item
+	 * @return the image of the table item
+	 */
+	private Image getImageForTableItem(SWTBotTableItem swtBotTableItem) {
+		List<Image> image = new ArrayList<>();
```

Comment:
This looks weird but I guess there was a reason :D

Reply:
I found a cleaner way to do it.